### DISCUSSION TITLE
Having a text-based format

### DISCUSSION THREAD

**Discussion by CodeDoctorDE** [LinwoodDev/Butterfly#770] [Ideas]
To have git compatibility, we need to think about having a format that is text-based as an alternative to the binary format based on a zip archive.

We could do it in two different ways:

## Adding a folder based setup

Instead of having directories inside an archive, we could write it directly inside folders.

Advantage: All assets are separated in an organized way.

Disadvantage: Handling of multiple files could be tricky. Additionally we need to have a root file to let the app know that this is a butterfly project directory.

## Having everything in a json file

Including all assets in a base64 format.

Advantage: One file for everything. You can easily download it and change it.
Disadvantage: Performance issues on large files. Images are very large if there are multiple

## Feedback

What do you think? What are your thoughts about this? Do you have a better idea for this?

**Comment by simono41** [comment]
I find the idea of ​​having everything in one json file more practical.
If you were to use it as a folder setup you would also have to take into account the differences between Windows and Unixiode filesystems.

**Reply by CodeDoctorDE** [reply]
Yeah that's the catch. Also on web it's not practical.
What do you think about a file extension .bflyt (bfly file - text based) or something like this?
Then the app can easily recognize the file format
